providers/openlineage/src/airflow/providers/openlineage/utils/spark.py

```
@@ -53,35 +53,66 @@ def _get_parent_job_information_as_spark_properties(context: Context) -> dict:
+            "timeoutInMillis": str(
+                int(tp.timeout * 1000)  # convert to milliseconds, as required by Spark integration
+            ),
+        }
+        if hasattr(tp, "compression") and tp.compression:
+            properties["compression"] = str(tp.compression)
+
+        if hasattr(tp.config.auth, "api_key") and tp.config.auth.get_bearer():
+            properties["auth.type"] = "api_key"
+            properties["auth.apiKey"] = tp.config.auth.get_bearer()
+
+        if hasattr(tp.config, "custom_headers") and tp.config.custom_headers:
+            for key, value in tp.config.custom_headers.items():
+                properties[f"headers.{key}"] = value
+        return properties
+
+    def _format_transport(props: dict, transport: dict, name: str | None):
+        for key, value in transport.items():
+            if name:
+                props[f"spark.openlineage.transport.transports.{name}.{key}"] = value
+            else:
+                props[f"spark.openlineage.transport.{key}"] = value
+        return props
 
-    if hasattr(transport.config, "custom_headers") and transport.config.custom_headers:
-        for key, value in transport.config.custom_headers.items():
-            properties[f"spark.openlineage.transport.headers.{key}"] = value
+    transport = get_openlineage_listener().adapter.get_or_create_openlineage_client().transport
 
-    return properties
+    if transport.kind == "composite":
+        http_transports = {}
+        for nested_transport in transport.transports:
+            if nested_transport.kind == "http":
+                http_transports[nested_transport.name] = _get_transport_information(nested_transport)
+        if len(http_transports) == 0:
+            log.info(
+                "OpenLineage transport type `composite` does not contain http transport. Skipping "
+                "injection of OpenLineage transport information into Spark properties.",
+            )
+            return {}
```

I think we should log a warning here if we encounter transports other than HTTP, instead of silently skipping them. We are in the place where only users that opted-in will be so we can show some extra logs. 

Why warning and not info like below? If we have a simple transport and it's not http, the events will not be delivered - it's easy to spot. With composite transport, users may expect that full translation to spark has happened, and they would receive events for their http endpoints, so it may be a bit misleading/not clear for them on why not all backends have received events.

---

Makes sense. Changed log level to warning.